code/game/objects/items/food/egg.dm

```diff
@@ -68,24 +69,22 @@ GLOBAL_VAR_INIT(chicks_from_eggs, 0)
 
 	var/turf/hit_turf = get_turf(hit_atom)
 	new /obj/effect/decal/cleanable/food/egg_smudge(hit_turf)
-	//Chicken code uses this MAX_CHICKENS variable, so I figured that I'd use it again here. Even this check and the check in chicken code both use the MAX_CHICKENS variable, they use independent counter variables and thus are independent of each other.
-	if(GLOB.chicks_from_eggs < MAX_CHICKENS) //Roughly a 1/8 (12.5%) chance to make a chick, as in Minecraft, with a 1/256 (~0.39%) chance to make four chicks instead.
-		var/chance = rand(0, 255)
-		switch(chance)
-			if(0 to 30)
-				new /mob/living/basic/chick(hit_turf)
-				GLOB.chicks_from_eggs++
-				visible_message(span_notice("A chick comes out of the cracked egg!"))
-			if(31)
-				var/spawned_chickens = min(4, MAX_CHICKENS - GLOB.chicks_from_eggs) // We don't want to go over the limit
-				visible_message(span_notice("[spawned_chickens] chicks come out of the egg! Jackpot!"))
-				for(var/i in 1 to spawned_chickens)
-					new /mob/living/basic/chick(hit_turf)
-					GLOB.chicks_from_eggs++
-
+	if(GLOB.chicks_from_eggs < MAX_CHICKENS && prob(chick_throw_prob))
+		spawn_impact_chick(hit_turf)
 	reagents.expose(hit_atom, TOUCH)
 	qdel(src)
 
+/// Spawn a baby chicken from throwing an egg
+/obj/item/food/egg/proc/spawn_impact_chick(turf/spawn_turf)
+	var/spawned_chickens = prob(97) ? 1 : min(4, MAX_CHICKENS - GLOB.chicks_from_eggs) // We don't want to go over the limit
+	if (spawned_chickens > 1) // Chicken jackpot!
```

Comment:
Should the `MAX_CHICKENS` check be baked into this proc instead? I figure something like:
```suggestion
/obj/item/food/egg/proc/spawn_impact_chick(turf/spawn_turf)
	var/remaining_spawnable_chicks = MAX_CHICKENS - GLOB.chicks_from_eggs
	if(remaining_spawnable_chicks <= 0)
		return
	var/spawned_chickens = prob(97) ? 1 : min(4, remaining_spawnable_chicks) // We don't want to go over the limit
	if (spawned_chickens > 1) // Chicken jackpot!
```

would work nicely enough instead of having to run this check before every invocation of this proc. we can have a `force` parameter if you want to bypass it but it shouldn't affect your watcher egg code anyways